### PR TITLE
drawContours: fix data types so that the function works even for huge images

### LINKED ISSUE #7458: drawContours fails for very large images
##### System information (version)
- OpenCV => 3.1
- Operating System / Platform => Windows 64 Bit
- Compiler => Visual Studio 2013
##### Detailed description

For very large images, the drawContours function does not produce the expected results.

See also #7449 – probably related. However, drawContours does not crash the program – it just produces an incorrect result.
##### Steps to reproduce

Please note that you'll need to have merged at least one of #7451 and #7452, because otherwise already the `findContours` call crashes the program:

```
cv::Mat largeImage(79925, 49955, CV_8UC1);
largeImage.setTo(0);

cv::circle(largeImage, cv::Point(largeImage.cols / 2, largeImage.rows / 2), 5000, 255, CV_FILLED);

std::vector<std::vector<cv::Point>> contours;
cv::findContours(largeImage, contours, cv::RETR_EXTERNAL, cv::CHAIN_APPROX_SIMPLE);

largeImage.setTo(0);
cv::drawContours(largeImage, contours, -1, 255, CV_FILLED);

cv::Mat smallImage;
cv::resize(largeImage, smallImage, cv::Size(499, 799));

cv::imwrite("smallImage.png", smallImage);
```

Result (not ok):
smallimage

However, this code does work as expected (when pasted right after the above snippet, so that `contours` has already been found):

```
std::vector<std::vector<cv::Point>> contours2 = contours;
for (auto& contour : contours2) {
    for (auto& point : contour) {
        point.x /= 100;
        point.y /= 100;
    }
}

cv::Mat smallImage2(smallImage.size(), CV_8UC1);
smallImage2.setTo(0);
cv::drawContours(smallImage2, contours2, -1, 255, CV_FILLED);
cv::imwrite("smallImage2.png", smallImage2);
```

Result (ok):
smallimage2

I'll try to find and fix the problem myself.

### PR DESCRIPTION
**Pull request by reunanen**
resolves #7458

### CODE REVIEW

File: modules/imgproc/src/drawing.cpp [hunk 1]
```diff
@@ -1195,6 +1203,10 @@ FillConvexPoly( Mat& img, const Point* v, int npts, const void* color, int line_
     while( ++y <= ymax );
 }
 
+bool fitsInInt32(XCoordinate x)
```
Comment (alalek) [inline modules/imgproc/src/drawing.cpp]: Please use `static` for local functions (or `static inline`).

Comment (reunanen) [inline modules/imgproc/src/drawing.cpp]: Fixed.

**Comment by vpisarev** [comment]
@terfendail, is this PR still relevant?

**Comment by terfendail** [comment]
I think no. Contours should now be drawn on huge images without issues since PR #7161 updated drawing.cpp to use int64 where necessary.
Also PR #7464 that tests contours drawing is already merged and doesn't fail so it looks like the issue is gone.

**Comment by reunanen** [comment]
Yep, seems to be a dupe of #7161. Wish I had known about it in the first place... Would have saved a lot of time. Well, my bad of course – should have looked around.

In line with @vpisarev's [comment](https://github.com/opencv/opencv/pull/7161#issuecomment-246991808), this PR would have left 32-bit builds to use `int` rather than `int64`. However, definitely not saying the `typedef` solution is great.

Anyway, I'm ok with #7161, so will close this one. Thanks!

**Pull request closed without merging**